decentraland/unity-explorer
# chore: optimize Avatar Visibility system

## What does this PR change?

- Caches the state of the avatar visibility to avoid changing it when not necessary
- Removes the setting of the local bound on every frame. Should be enough to do it [here](https://github.com/decentraland/unity-explorer/blob/d2a9f9fdebf7ac7b3236c280dc1295b970032aca/Explorer/Assets/DCL/AvatarRendering/AvatarShape/ComputeShader/ComputeShaderSkinning.cs#L205)

For 30 avatars, gos from this

![image](https://github.com/user-attachments/assets/4bf79ced-943f-46b0-8e5a-9909f60e565e)
 
to this

![image](https://github.com/user-attachments/assets/ef4176d8-8b3b-4c62-8dce-2bd7afe5544d)


## How to test the changes?

1. Launch the explorer
2. Go to Goerli, to the avatar swap scene (`75,-7`). Should work as expected
3. Instantiate 30 avatars. They should all be visible
4. Change to first person. Your avatar should dissapear

## Our Code Review Standards

https://github.com/decentraland/unity-renderer/blob/master/docs/code-review-standards.md
